alacritty_terminal/src/selection.rs

```diff
@@ -193,6 +193,22 @@ impl Selection {
         }
     }
 
+    /// Check whether selection contains any point in a given range.
+    pub fn intersects_range(&self, range: RangeInclusive<usize>) -> bool {
+        let (mut start, mut end) = (self.region.start, self.region.end);
+        if Self::points_need_swap(start.point, end.point) {
+            mem::swap(&mut start, &mut end);
+        }
+
+        let (start, end) = (start.point.line, end.point.line);
+
+        let (range_start, range_end) = range.into_inner();
+
+        (start >= range_end && end <= range_end)
+            || (start >= range_start && end <= range_start)
+            || (start >= range_start && start <= range_end)
```

Comment:
I'm gonna sound a bit silly if this is not true, but wouldn't the following condition be sufficient:

```
range.start >= selection.end && range.end <= selection.start
```

My thinking here is that we know for a fact that our selection is well structured. So we know the start is above the end when we swap points. So by knowing that the range end is below the range start, and knowing that the range start is above the selection end and the range end is below the selection start, we know that they must intersect?

Reply:
I don't get what are you talking about here. You have 3 patterns

- selection starts above, ends below range_end
- selection starts below and ends above range_start
- selection between range start and end

I'm not sure that i get your suggestion here, since it's not correct for case.

`range.start` is below `selection.end` and `selection.start` is above `range.end`(range inside selection).

Reply:
I think my start/end are swapped around. In my example the start is the top of the range and selection and the end is the bottom of the range and selection.

For the case you've mentioned, it would work if that's the case. You'd just have to swap them around to match that.